File: BE/living-server/src/main/java/com/living/hana/entity/Contract.java
package com.living.hana.entity;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

import java.math.BigDecimal;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Contract {
    
    private Long id;
    private Long userId;
    private Long unitId;
    private String contractNumber;
    private String startDate;
    private String endDate;
    private BigDecimal deposit; // 보증금
    private BigDecimal monthlyRent; // 월세
    private String status; // ACTIVE, EXPIRED, TERMINATED
    private String moveInDate; // 입주일 (YYYY-MM-DD)
    private Integer paymentDay; // 납부일 (1-31일, 입주일 기준)
    private String contractImageUrl; // 계약서 이미지 URL
    private String createdAt;
    private String updatedAt;
}

File: FE/living-erp/src/types/contract.ts
// 계약 관련 타입 정의
export interface Contract {
  id: number;
  userId: number;
  unitId: number;
  startDate: string;
  endDate: string;
  deposit: number;
  monthlyRent: number;
  maintenanceFee: number;
  currentAddress?: string;
  status: 'ACTIVE' | 'EXPIRED' | 'TERMINATED';
  createdAt: string;
  updatedAt: string;
}

// 계약 생성 요청 타입
export interface CreateContractRequest {
  userId: number;
  unitId: number;
  startDate: string; // YYYY-MM-DD 형식
  endDate: string;   // YYYY-MM-DD 형식
  deposit: number;
  monthlyRent: number;
  maintenanceFee: number;
  currentAddress?: string;
  paymentDay: number; // 월세/관리비 통합 납부일 (1-31)
  moveInDate: string; // 입주일 (YYYY-MM-DD 형식)
}

// 계약 생성 응답 타입
export interface CreateContractResponse {
  success: boolean;
  data: Contract;
  message: string;
}

// 계약 폼 데이터 타입
export interface ContractFormData {
  startDate: string;
  endDate: string;
  deposit: string;
  monthlyRent: string;
  maintenanceFee: string;
  currentAddress: string;
  paymentDay: string;
  moveInDate: string;
}
